relays/messages/src/message_race_strategy.rs

```diff
@@ -266,19 +270,37 @@ impl<
 	) {
 		let nonce = nonces.latest_nonce;
 
+		// if **some** of nonces that we have selected to submit already present at the
+		// target chain => select new nonces
 		let need_to_select_new_nonces = race_state
 			.nonces_to_submit()
-			.map(|nonces| *nonces.end() <= nonce)
+			.map(|nonces| nonce >= *nonces.start())
 			.unwrap_or(false);
-		if need_to_select_new_nonces {
+		if need_to_select_new_nonces && race_state.nonces_to_submit().is_some() {
```

Comment:
That's true, thanks for spotting. I've added this to avoid spam in logs, but you're right - the flag can't be `true` unless we have something to "clear"